connect-location/src/main/java/com/ifttt/location/ConnectLocation.java

```diff
@@ -104,11 +167,31 @@ public void onStateChanged(
 
         if (currentState == ConnectButtonState.Enabled) {
             setUpPolling();
+            if (permissionCallback != null && !permissionChecker.hasLocationPermission()) {
+                permissionCallback.onLocationPermissionRequested();
+            }
         }
     }
 
     @Override
     public void onError(ErrorResponse errorResponse) {
         // No-op
     }
+
+    interface LocationPermissionCallback {
+        void onLocationPermissionRequested();
```

Comment:
Could consider providing the permission string or the array here, so that developers can just pass that into the activity's request permission method.